cnvlib/vary.py

```diff
@@ -252,19 +256,19 @@ def _parse_records(vcf_reader, sample_id, normal_id, skip_reject):
         logging.info('Filtered out %d records', cnt_reject)
 
 
-def _extract_genotype(sample):
+def _extract_genotype(sample, ignore_gt):
     if "DP" in sample.data._fields:
         depth = sample.data.DP
     else:
         # SV, probably
         depth = alt_count = None
     if sample.is_het:
         zygosity = 0.5
-    elif sample.gt_type == 0:
+    elif not ignore_gt and sample.gt_type == 0:
         zygosity = 0.0
     else:
         zygosity = 1.0
-    alt_count = (_get_alt_count(sample) if sample.gt_type else 0.0)
+    alt_count = _get_alt_count(sample) if ignore_gt or sample.gt_type else 0.0
```

Comment:
If we just remove the original `if sample.gt_type else 0.0`, and the user can set the `zygosity` values to 0.5 after parsing, does that fix the problem without introducing a new option?